examples/lib/versionManagement.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * Functions for getting version information.
+ * Note: these functions are not used by the service worker to check for updates;
+ * the service worker updates whenever sw.js has changed.
+ * sw.js is changed when grunt replace:version is run. This task is run during
+ * grunt build, serve, and productions tasks.
+ */
+
+/**
+ * Get the current version number from package.json on the homepage.
+ * @param {function} callback The function that uses the version number.
+ */
+function getLatestVersionNumber(callback) {
+  // Get the homepage reference from the local package.json.
+  var homepage = require('../../package.json').homepage;
+  var request = new XMLHttpRequest();
+  request.onreadystatechange = function() {
+    if(request.readyState == 4 && request.status == 200) {
+      var response = JSON.parse(this.responseText);
+      var latestVersionNumber = response.version;
+
+      // Do something with the version number using a callback function.
```

Comment:
Is this callback optional?

Reply:
It's needed to use the version number fetched, but it's probably still good practice to check if it is defined using `if (callback)`.